cdc_uart: add RTS and DTR pins

cyber-murmel opened the pull request:
I want to use the pico probe firmware as a debugger for another platform, that can be flashed via UART, similar to Arduino. My tooling uses the RTS and DTR lines as reset and boot mode.

This PR adds these outgoing flow control lines.

P33M commented:
This PR is incompatible with a) branch https://github.com/raspberrypi/picoprobe/tree/debugprobe (which I have forgotten to merge) and b) will unconditionally drive outputs that may be used for other purposes. Please rebase against debugprobe, and consider carefully under which conditions it should be considered safe to connect DTR/RTS.

cyber-murmel commented:
I added the RTS and DTR pin definition only to the example board config for now and added ifdefs to the c module.
Maybe the config for the pico could have RTS and DTR on GP6 and GP7.

rewolff commented:
@P33M it is kind of difficult to "see" that the pins are actually used from the embedded CDC side. The OS will send updates to the signals even when just connecting and then activating say minicom. 

So, in general, the only remaining option is to do tricks with defines to do it compile-time, or to develop a mechanism to dynamically configure the uart. Here a special packet to the DAP port could work. Still kind of hacky. I'd suggest to go for the "compile time option" @cyber-murmel  suggests.
